crates/matrix-sdk/src/sliding_sync/builder.rs

```diff
@@ -190,10 +190,24 @@ impl SlidingSyncBuilder {
         self
     }
 
-    /// Sets custom timeouts for the sliding sync requests.
-    pub fn with_timeouts(mut self, polling_timeout: Duration, network_timeout: Duration) -> Self {
-        self.polling_timeout = polling_timeout;
-        self.network_timeout = network_timeout;
+    /// Sets a custom timeout duration for the
+    /// sliding sync polling endpoint.
+    ///
+    /// This is the timeout that appears in the sliding sync request's query
```

Comment:
Perhaps we can clarify this a bit more, taking inspiration from the spec:

> The maximum time to wait, in milliseconds, before returning this request. If no events (or other data) become available before this time elapses, the server will return a response with empty fields.

That's from the point of view of the server, but rewording it for it to be from the point of view of the client shouldn't be too hard.